src/relocation.rs

```diff
@@ -122,10 +126,127 @@ impl RelocatePayload {
     }
 }
 
-pub fn compute_destination(relocated_name: &XorName, trigger_name: &XorName) -> XorName {
+#[cfg(not(feature = "mock_base"))]
+pub fn compute_destination(
+    _src_prefix: &Prefix<XorName>,
+    relocated_name: &XorName,
+    trigger_name: &XorName,
+) -> XorName {
+    compute_destination_without_override(relocated_name, trigger_name)
+}
+
+#[cfg(feature = "mock_base")]
+pub fn compute_destination(
+    src_prefix: &Prefix<XorName>,
+    relocated_name: &XorName,
+    trigger_name: &XorName,
+) -> XorName {
+    self::overrides::get(
+        src_prefix,
+        compute_destination_without_override(relocated_name, trigger_name),
+    )
+}
+
+fn compute_destination_without_override(
+    relocated_name: &XorName,
+    trigger_name: &XorName,
+) -> XorName {
     let mut buffer = [0; 2 * XOR_NAME_LEN];
     buffer[..XOR_NAME_LEN].copy_from_slice(&relocated_name.0);
     buffer[XOR_NAME_LEN..].copy_from_slice(&trigger_name.0);
 
     XorName(crypto::sha3_256(&buffer))
 }
+
+#[cfg(feature = "mock_base")]
+mod overrides {
+    use crate::{Prefix, XorName};
+    use std::{
+        cell::{Cell, RefCell},
+        collections::HashMap,
+    };
+
+    /// Mechanism for overriding relocation destinations. Useful for tests.
+    pub struct Overrides;
+
+    impl Overrides {
+        /// Create new instance of relocation overrides. There can be only one per thread.
```

Comment:
`shall be only one instance per thread`?
same to the below

Reply:
I think "can" implies that it is actually impossible to create more than one instance (because it will panic otherwise), while "shall" to me sounds like you are just advised to not create more than one.